File: Native/iOS/CT-PL-Pn-Integration/CT-PL-Pn-Integration/DeepLink/DeepLinkModalManager.swift
import Foundation
import Combine

/// Holds the deeplink data received from a push notification tap and exposes it
/// to the UI so a modal can be shown. CleverTap and Plotline both deliver their
/// deeplink payload here (see AppDelegate). Mirrors the Android
/// `DeepLinkModalManager`.
final class DeepLinkModalManager: ObservableObject {

    static let shared = DeepLinkModalManager()

    struct DeepLinkData {
        let source: String
        let deepLink: String
    }

    @Published var data: DeepLinkData?

    private init() {}

    func present(source: String, deepLink: String) {
        data = DeepLinkData(source: source, deepLink: deepLink)
    }

    func dismiss() {
        data = nil
    }
}

File: Native/Android/CTPLPnIntegration/app/src/main/java/com/vamsi/ct_pl_pnintegration/deeplink/DeepLinkModalManager.kt
package com.vamsi.ct_pl_pnintegration.deeplink

import kotlinx.coroutines.flow.MutableStateFlow

/**
 * Holds the deeplink data received from a push notification tap and exposes it
 * to the UI so a modal can be shown. CleverTap and Plotline both deliver their
 * deeplink payload here (see CTPLApplication and MainActivity).
 */
object DeepLinkModalManager {

    data class DeepLinkData(
        val source: String,
        val deepLink: String
    )

    val deepLink = MutableStateFlow<DeepLinkData?>(null)

    fun present(source: String, deepLink: String) {
        this.deepLink.value = DeepLinkData(source, deepLink)
    }

    fun dismiss() {
        deepLink.value = null
    }
}